chia/util/condition_tools.py

```diff
@@ -71,6 +71,45 @@ def agg_sig_additional_data(agg_sig_data: bytes) -> Dict[ConditionOpcode, bytes]
     return ret
 
 
+def make_aggsig_final_message(
+    opcode: ConditionOpcode,
+    msg: bytes,
+    spend: Union[Coin, Spend],
+    agg_sig_additional_data: Dict[ConditionOpcode, bytes],
+) -> bytes:
+    if isinstance(spend, Coin):
+        parent_coin_info = bytes(spend.parent_coin_info)
+        puzzle_hash = bytes(spend.puzzle_hash)
+        amount = int_to_bytes(spend.amount)
+        coin_id = bytes(spend.name())
+    elif isinstance(spend, Spend):
+        parent_coin_info = spend.parent_id
+        puzzle_hash = spend.puzzle_hash
+        amount = int_to_bytes(spend.coin_amount)
+        coin_id = spend.coin_id
+    else:
+        raise ValueError(f"Expected Coin or Spend, got {type(spend)}")
+
+    if opcode == ConditionOpcode.AGG_SIG_PARENT:
```

Comment:
Python 3.10 supports `match` but a python trick I like is to use a `dict` rather than a large if/else branch.

```
addendum = dict(
   ConditionOpcode.AGG_SIG_PARENT = lambda coin: coin.parent_info,
   ...
)[opcode](coin)
```

It's probably not as readable. Don't know offhand if it's faster or slower. But it's definitely more concise and symmetric.

Reply:
Thanks, done.